torch/functional.py

```diff
@@ -66,3 +66,59 @@ def unbind(tensor, dim=0):
         dim (int): dimension to remove.
     """
     return tuple(tensor.select(dim, i) for i in _range(tensor.size(dim)))
+
+
+def btriunpack(LU_data, LU_pivots, mode='E', dtype=None):
+    """Unpacks the data and pivots from a batched LU factorization (btrifact) of a tensor.
+
+    Returns a tuple indexed by:
+      0: The pivots.
+      1: The L tensor.
+      2: The U tensor.
+
+    The mode flag indicates what should be unpacked:
+      'E' unpacks everything (data and pivots)
+      'D' unpacks just the data
+      'P' unpacks just the pivots
+
+    Arguments:
+        LU_data (Tensor): The packed LU factorization data.
+        LU_pivots (Tensor): The packed LU factorization pivots.
+        mode (str): The mode indicating what should be unpacked.
+    """
+    assert mode in ('E', 'D', 'P')
+
+    if mode in ('E', 'D'):
+        nBatch, sz, _ = LU_data.size()
+        I_U = torch.triu(torch.ones(sz, sz)).unsqueeze(0).expand(nBatch, sz, sz).type_as(LU_data).byte()
```

Comment:
Convert the type before expanding the tensor. It's going to use `batch_size` less memory.